core/domain/suggestion_registry.py

```diff
@@ -654,6 +724,48 @@ def _get_skill_difficulty(self):
         """Returns the suggestion's skill difficulty."""
         return self.change.skill_difficulty
 
+    def get_all_html_content_strings(self):
+        """Gets all html content strings used in this suggestion.
+
+        Returns:
+            list(str). The list of html content strings.
+        """
+        html_string_list = []
+        interaction_object = (
+            state_domain.InteractionInstance.from_dict(
+                self.change.question_dict['question_state_data'][
+                    'interaction']))
+        html_string_list = (
+            html_string_list +
+            interaction_object.get_all_html_content_strings())
+        written_translations_dict = (
+            self.change.question_dict['question_state_data'][
+                'written_translations'])
+        written_translations_object = (
+            state_domain.WrittenTranslations.from_dict(
+                written_translations_dict))
+        html_string_list = (
+            html_string_list +
+            written_translations_object.get_all_html_content_strings())
+        return html_string_list
+
+    def convert_html_in_suggestion(self, conversion_fn):
+        """Checks for HTML fields in the suggestion and converts it according
+        to the conversion function.
+
+        Args:
+            conversion_fn: function. The function to be used for converting the
+                HTML.
+        Returns:
+            suggestion change. QuestionSuggestionChange. The converted
+            Suggestion change.
+        """
+        self.change.question_dict['question_state_data'] = (
+            state_domain.State.convert_html_fields_in_state(
+                self.change.question_dict['question_state_data'],
+                conversion_fn))
+        return question_domain.QuestionSuggestionChange(self.change.to_dict())
```

Comment:
Ditto as above why this object property is returning dict?

Reply:
changed to class_method

Reply:
Any specific reason to change this to clasmethod not as property method?

Reply:
same as this [comment](https://github.com/oppia/oppia/pull/9390#discussion_r432852227)